### PR TITLE
Allow article cover images to be deleted

### LINKED ISSUE #1691: allow clearing the cover image on an article post
currently the API doesn't support removing an image from an article post, all you can do is replace your existing image with a new one.

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1691

#### What's this PR do?
- Removes an article cover image if `article_cover` is not sent with a patch request.
- Prevents an exception from occurring if `article_cover` in the patch request is a URL instead of an uploaded file (which happens on master when editing an existing article but leaving the cover as is).

#### How should this be manually tested?
- Edit an article with a cover image.  Change the text but not the cover image.  Save.  You should not see an exception when the post detail page reloads, and the cover image should still exist.
- Edit the article again.  Remove the cover image.  Save.  You should not see an exception when the post detail page reloads, and the cover image should be gone.

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks great! taking a look at the code now.

File: channels/views/posts_test.py [hunk 1]
```diff
@@ -783,6 +783,22 @@ def test_update_article_cover(
...
+    article = Article.objects.get(post__post_id=resp.json()["id"])
+    assert image_url.endswith(article.cover_image.url)
+
+    # Remove the image
+    resp = user_client.patch(url, {"article_content": "foo"}, format="json")
+    assert resp.status_code == status.HTTP_200_OK
+    article = Article.objects.get(post__post_id=resp.json()["id"])
+    assert article.cover_image.name == ""
+    assert article.cover_image_small.name == ""
+
```
Comment (aliceriot) [inline channels/views/posts_test.py]: do you think we should also add a test making sure you can issue a PATCH request with a URL string for the cover image?

Comment (mbertrand) [reply channels/views/posts_test.py]: That is tested on lines 786-793, but I can put it in a separate test if you prefer

Comment (aliceriot) [reply channels/views/posts_test.py]: gosh I should really read more closely! haha

**Pull request merged**